engine/model_test.go

```diff
@@ -294,3 +371,316 @@ func (s *BackendSuite) TestServerFromJSON(c *C) {
 
 	c.Assert(out, DeepEquals, e)
 }
+
+func (s *BackendSuite) TestNewTLSSettings(c *C) {
+	tcs := []struct {
+		S TLSSettings
+		C *tls.Config
+	}{
+		// Make sure defaults are set as expected
+		{
+			S: TLSSettings{},
+			C: &tls.Config{
+				MinVersion: tls.VersionTLS10,
+				MaxVersion: tls.VersionTLS12,
+
+				SessionTicketsDisabled:   false,
+				PreferServerCipherSuites: false,
+				CipherSuites: []uint16{
+					tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+					tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+
+					tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+					tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+
+					tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
+					tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+
+					tls.TLS_RSA_WITH_AES_256_CBC_SHA,
+					tls.TLS_RSA_WITH_AES_128_CBC_SHA,
+				},
+
+				InsecureSkipVerify: false,
+			},
+		},
+		{
+			S: TLSSettings{
+				MinVersion: "VersionTLS11",
+				MaxVersion: "VersionTLS12",
+
+				SessionTicketsDisabled:   false,
+				PreferServerCipherSuites: true,
+
+				CipherSuites: []string{
+					"TLS_RSA_WITH_RC4_128_SHA",
+					"TLS_RSA_WITH_3DES_EDE_CBC_SHA",
+					"TLS_ECDHE_ECDSA_WITH_RC4_128_SHA",
+					"TLS_ECDHE_RSA_WITH_RC4_128_SHA",
+					"TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA",
+
+					"TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
+					"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+					"TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
+					"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
+					"TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA",
+					"TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA",
+					"TLS_RSA_WITH_AES_256_CBC_SHA",
+					"TLS_RSA_WITH_AES_128_CBC_SHA",
+				},
+
+				InsecureSkipVerify: false,
+
+				SessionCache: TLSSessionCache{
+					Type: "LRU",
+					Settings: &LRUSessionCacheSettings{
+						Capacity: 12,
+					},
+				},
+			},
+			C: &tls.Config{
+				MinVersion: tls.VersionTLS11,
+				MaxVersion: tls.VersionTLS12,
+
+				SessionTicketsDisabled:   false,
+				PreferServerCipherSuites: true,
+				CipherSuites: []uint16{
+					tls.TLS_RSA_WITH_RC4_128_SHA,
+					tls.TLS_RSA_WITH_3DES_EDE_CBC_SHA,
+					tls.TLS_ECDHE_ECDSA_WITH_RC4_128_SHA,
+					tls.TLS_ECDHE_RSA_WITH_RC4_128_SHA,
+					tls.TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA,
+
+					tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+					tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+
+					tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+					tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+
+					tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
+					tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+
+					tls.TLS_RSA_WITH_AES_256_CBC_SHA,
+					tls.TLS_RSA_WITH_AES_128_CBC_SHA,
```

Comment:
Can you keep the cipher suite spacing the same between `C` and `S`? I think that would make it easier to tell at a glance that they are the same.